Replace 'query error' with 'request error'

benjie opened the pull request:
This is a specific change from #777 applied throughout the spec.

The term "query error" feels limited to GraphQL query operations; however the error could be raised from mutations or subscriptions too, in fact it can also be raised when it's not clear what operation is to be executed (e.g. when there's a document with multiple operations, but no `operationName` was provided). As such, the error does not relate specifically to "query", or even "operation", but to the entire GraphQL request.

Relates to https://github.com/graphql/graphql-wg/issues/546

leebyron commented:
For posterity from live discussion:

Agreement on this change - a further improvement would be to at some point describe what a "request error" is

benjie commented:
I've added a definition for request errors and linked it from the first time it's mentioned (I'm not sure if it should be linked from each place?). I've also reviewed the content in section 7 and updated some of the uses of "operation" to "request" where they may apply before the operation can even be determined.

benjie commented on spec/Section 7 -- Response.md:
```diff
@@ -8,19 +8,26 @@ request.
+A response to a GraphQL request must be a map.
 
-If the operation encountered any errors, the response map must contain an
+If the request encountered any errors, the response map must contain an
 entry with key `errors`. The value of this entry is described in the "Errors"
 section. If the operation completed without encountering any errors, this entry
 must not be present.
```
Note the request might encounter an error before the operation can be determined (for example if `operationName` is not given but the document contains multiple operations, or of the document contains no operations), hence 'If the request encountered any errors.' However the final sentence is still correct - it does indeed relate to the **operation** completing without errors - for the operation to even begin there must be no errors before this point. Since it's valid both ways (and there's no ambiguity) I left this final sentence unedited.

leebyron commented:
@benjie - please let me know if my additions here look good to you

benjie commented:
This looks good to me 👍

glasser commented on spec/Section 3 -- Type System.md:
```diff
@@ -1514,10 +1516,10 @@ type of an Object or Interface field.
-with names not defined by a field of this input object type, otherwise an error
-must be thrown.
+with names not defined by a field of this input object type, otherwise a
+response error must be raised.
```
@benjie I noticed this term "response error" which is used only once in the spec and was introduced in this PR (which was the PR that introduced the term "request error"). Was this a mistake and should also say "request error"?

benjie replied:
Indeed, I suspect it should either be `request error` or `field error`. I'd have to read it carefully to figure out which - I'll add this to my list to review. Thanks for spotting!

benjie replied:
It should have been `request error` indeed. Thanks for spotting!
